### PR TITLE
libaesgm: add recipe

### PR DESCRIPTION
**Pull request by toge**
Specify library name and version:  **libaesgm/***

I want to create [PDF-Writer](https://github.com/galkahana/PDF-Writer) recipe.
It depends on libaesgm. 
While libaesgm official site has been down. http://gladman.plushost.co.uk/oldsite/AES/index.php, 
I found [mirror site](https://github.com/xmake-mirror/libaesgm) which adopt pdfwriter's modifications.

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/libaesgm/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+from conan.tools.cmake import CMake, CMakeDeps, CMakeToolchain, cmake_layout
+import os
+
+
+required_conan_version = ">=1.53.0"
+
+class LibaesgmConan(ConanFile):
+    name = "libaesgm"
+    description = "Library implementation of AES (Rijndael) cryptographic methods"
+    license = "BSD"
```
Comment (uilianries) [inline recipes/libaesgm/all/conanfile.py:12]:
```suggestion
    license = "LicenseRef-libaesgm-BSD"
```

It looks like BSD-2, but is more simplified, so it's not an official SPDX license. Let's follow https://github.com/conan-io/conan-center-index/blob/master/docs/adding_packages/conanfile_attributes.md#license-attribute

File: recipes/libaesgm/all/conandata.yml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+sources:
+  "2013.1.1":
+    url: "https://github.com/xmake-mirror/libaesgm/archive/refs/tags/2013.1.1.tar.gz"
```
Comment (uilianries) [inline recipes/libaesgm/all/conandata.yml:3]: The original page should be http://gladman.plushost.co.uk/oldsite/AES/index.php but it no longer exists. Both Fedora and other Linux distros are using an internal copy of their source files. So using this mirror look fine.

Extra info: https://packages.altlinux.org/en/p9/srpms/libaesgm/

**Review by uilianries — COMMENTED** [review]
@toge thank you for your contribution. I sent your CMake file with few changes to the upstream: https://github.com/xmake-mirror/libaesgm/pull/1

File: recipes/libaesgm/all/conandata.yml (hunk 2 above)
Comment (toge) [reply recipes/libaesgm/all/conandata.yml:3]: @uilianries 
Thanks a lot for your detail information!

**Review by uilianries — APPROVED** [review]
LGTM

**Review by RubenRBS — APPROVED** [review]

**Pull request merged**